metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+                )
+        elif self.attributes["events"]:
+            # events attribute supports the following formats -
+            #     1. events=[{'name': 'table.prod_db.members',
+            #               'parameters': {'alpha': 'member_weight'}},
+            #                {'name': 'table.prod_db.metadata',
+            #               'parameters': {'beta': 'grade'}}]
+            if isinstance(self.attributes["events"], list):
+                for event in self.attributes["events"]:
+                    if is_stringish(event):
+                        self.triggers.append({"name": str(event)})
+                    elif isinstance(event, dict):
+                        if "name" not in dict(event):
+                            raise MetaflowException(
+                                "One or more events in *events* attribute for "
+                                "*@trigger* are missing the *name* key."
+                            )
+                        self.triggers.append(event)
+                    else:
+                        raise MetaflowException(
+                            "One or more events in *events* attribute in *@trigger* "
+                            "decorator have an incorrect format. Supported format "
+                            "is dictionary - \n"
+                            "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                            "'beta'}}, {'name': 'bar', 'parameters': "
+                            "{'gamma': 'kappa'}}])"
+                        )
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *events* attribute in *@trigger* decorator. "
+                    "Supported format is list - \n"
+                    "@trigger(events=[{'name': 'foo', 'parameters': {'alpha': "
+                    "'beta'}}, {'name': 'bar', 'parameters': "
+                    "{'gamma': 'kappa'}}])"
+                )
+
+        if not self.triggers:
+            raise MetaflowException("No event(s) specified in *@trigger* decorator.")
+
+        # same event shouldn't occur more than once
+        names = [x["name"] for x in self.triggers]
+        if len(names) != len(set(names)):
+            raise MetaflowException(
+                "Duplicate event names defined in *@trigger* decorator."
+            )
+
+        self.options = self.attributes["options"]
+
+        # TODO: Handle scenario for local testing using --trigger.
+
+
+class TriggerOnFinishDecorator(FlowDecorator):
+    """
+    Specifies the flow(s) that this flow depends on.
+
+    ```
+    @trigger_on_finish(flow='FooFlow')
+    ```
+    or
+    ```
+    @trigger_on_finish(flows=['FooFlow', 'BarFlow'])
+    ```
+    This decorator respects the @project decorator and triggers the flow
+    when upstream runs within the same namespace complete successfully
+
+    Additionally, you can specify project aware upstream flow dependencies
```

Comment:
this is a new concept not documented so would add examples:
`project.prod.flow_name` for the production branch or `project.[prod|test].branch_name.flow_name` or `project.user.user_name.flow_name` or something like that.

Reply:
We will add examples in the documentation.